contracts/crosschain/ERC7786Aggregator.sol

```diff
@@ -0,0 +1,278 @@
+        } else {
+            // if already executed, revert
+            if (tracker.executed) revert ERC7786AggregatorAlreadyExecuted();
+        }
+
+        // Parse payload
+        (, string memory originalSender, string memory receiver, bytes memory unwrappedPayload) = abi.decode(
+            payload,
+            (uint256, string, string, bytes)
+        );
+
+        // If ready to execute, and not yet executed
+        if (tracker.countReceived >= getThreshold()) {
+            // prevent re-entry
+            tracker.executed = true;
+
+            bytes memory call = abi.encodeCall(
+                IERC7786Receiver.executeMessage,
+                (sourceChain, originalSender, unwrappedPayload, attributes)
+            );
+            // slither-disable-next-line reentrancy-no-eth
+            (bool success, bytes memory returndata) = receiver.parseAddress().call(call);
+
+            if (
+                success &&
+                returndata.length >= 32 &&
+                abi.decode(returndata, (bytes4)) == IERC7786Receiver.executeMessage.selector
+            ) {
+                emit ExecutionSuccess(id);
+            } else {
+                // rollback to enable retry
+                tracker.executed = false;
+                emit ExecutionFailed(id);
+            }
+        }
+
+        return IERC7786Receiver.executeMessage.selector;
+    }
+
+    // =================================================== Getters ===================================================
+
+    function getGateways() public view virtual returns (address[] memory) {
+        return _gateways.values();
+    }
+
+    function getThreshold() public view virtual returns (uint8) {
+        return _threshold;
+    }
+
+    function getRemoteRouter(string calldata caip2) public view virtual returns (string memory) {
+        string memory router = _remoteRouters[caip2];
+        if (bytes(router).length == 0) revert ERC7786AggregatorRemoteNotRegistered(caip2);
+        return router;
+    }
+
+    // =================================================== Setters ===================================================
+
+    function addGateway(address gateway) public virtual onlyOwner {
+        _addGateway(gateway);
+    }
+
+    function removeGateway(address gateway) public virtual onlyOwner {
+        _removeGateway(gateway);
+    }
+
+    function setThreshold(uint8 newThreshold) public virtual onlyOwner {
+        _setThreshold(newThreshold);
+    }
+
+    function registerRemoteRouter(string memory caip2, string memory router) public virtual onlyOwner {
+        _registerRemoteRouter(caip2, router);
+    }
+
+    function pause() public virtual onlyOwner {
+        _pause();
+    }
+
+    function unpause() public virtual onlyOwner {
+        _unpause();
+    }
+
+    /// @dev Recovery method in case ether is ever received through {executeMessage}
+    function drainEth(address payable to) public virtual onlyOwner {
```

Comment:
```suggestion
    /// @dev Recovery method in case value is ever received through {executeMessage}
    function sweep(address payable to) public virtual onlyOwner {
```